botocore/data/s3/2006-03-01/waiters-2.json

```diff
@@ -12,6 +12,16 @@
           "state": "success"
         },
         {
+          "expected": 301,
+          "matcher": "status",
+          "state": "success"
+        },
+        {
+          "expected": 403,
```

Comment:
Yeah, I'm on the fence with this one as well.

Technically it's correct.  An auth error could mean the bucket exists but you can't access it.  However, in practice  I think the most common workflow is:

```
create_bucket()        # 1.
wait_for_it_to_exist() # 2.
operate_on_bucket()    # 3.
```

In which case if something happens between (1) and (2) where you can no longer access the bucket, I think this should be an error rather than a waiter success.